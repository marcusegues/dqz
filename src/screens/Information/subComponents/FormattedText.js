// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
import { moderateScale } from '../../../styles/Scaling';
import { type } from '../../../styles/fonts';
import { MAIN_BLACK } from '../../../styles/colors';

type FormattedTextProps = {
  text: string,
  bold?: boolean,
  style?: {},
};

export const FormattedText = ({ text, style, bold }: FormattedTextProps) => (
  <Text
    style={{
      color: MAIN_BLACK,
      fontFamily: bold ? type.bold : type.light,
      lineHeight: moderateScale(21),
      ...style,
    }}
  >
    {text}
  </Text>
);

FormattedText.defaultProps = {
  style: {},
  bold: false,
};
