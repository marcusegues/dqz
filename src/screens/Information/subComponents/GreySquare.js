import React from 'react';
import { Text, View } from 'react-native';
import { moderateScale } from '../../../styles/Scaling';

const ownStyles = {
  container: {
    width: 64,
    height: 64,
    backgroundColor: '#D8D8D8',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontFamily: 'roboto_regular',
    color: '#898989',
    fontSize: moderateScale(14),
  },
};

const GreySquare = ({ number }) => (
  <View style={ownStyles.container}>
    <Text style={ownStyles.text}>{number}</Text>
  </View>
);

export default GreySquare;
