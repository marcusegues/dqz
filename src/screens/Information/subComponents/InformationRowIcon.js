// @flow
import React from 'react';
// $FlowFixMe
import { Image } from 'react-native';

const ownStyles = {
  image: {
    width: 64,
    height: 64,
    alignSelf: 'center',
  },
};

type InformationRowIconProps = {
  source: string,
};

export const InformationRowIcon = ({ source }: InformationRowIconProps) => (
  <Image source={source} style={ownStyles.image} resizeMode="contain" />
);
