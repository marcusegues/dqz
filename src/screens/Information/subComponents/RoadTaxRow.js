// @flow
import React from 'react';
// $FlowFixMe
import { Image, View } from 'react-native';
import { CardRowText } from '../../../components/QuestionAnswer/Cards/subcomponents/CardRowText';
import { Row } from '../../../components/Rows/Row';
import { AmountIcon } from '../../../components/General Components/GreyBox/configured/AmountIcon';
import type { Children } from '../../../types/generalTypes';

const ownStyles = pendant => ({
  innerContainer: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 16,
  },
  amountInputContainer: {
    flexDirection: pendant ? 'column' : 'row',
    alignItems: 'flex-end',
  },
  categoryTitle: {
    alignSelf: 'flex-start',
    width: pendant ? '60%' : '62%',
  },
});

type RoadTaxRowProps = {
  borderTop?: boolean,
  borderBottom?: boolean,
  children?: Children,
  source: string,
  imageStyle: {},
  title: string,
  amount: number,
  pendant?: boolean,
};

export const RoadTaxRow = ({
  borderTop,
  borderBottom,
  source,
  imageStyle,
  title,
  pendant,
  amount,
  children,
}: RoadTaxRowProps) => (
  <Row
    width="100%"
    borderTop={borderTop}
    borderBottom={borderBottom}
    styles={{ marginBottom: pendant ? 16 : 0 }}
  >
    <View style={ownStyles(pendant).categoryTitle}>
      <CardRowText text={title} style={{ marginVertical: 16 }} />
    </View>
    <View style={ownStyles().innerContainer}>
      <Image source={source} resizeMode="contain" style={imageStyle} />
      <View style={ownStyles(pendant).amountInputContainer}>
        <AmountIcon amount={amount} currency="CHF" />
        {children}
      </View>
    </View>
  </Row>
);

RoadTaxRow.defaultProps = {
  borderTop: false,
  children: [],
  pendant: false,
  borderBottom: true,
};
