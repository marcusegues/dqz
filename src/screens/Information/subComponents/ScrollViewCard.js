// @flow
import React from 'react';
// $FlowFixMe
import { View, ScrollView } from 'react-native';
import globalStyles from '../../../styles/globalStyles';

const ownStyles = {
  container: {
    flex: 1,
    width: '90%',
    alignSelf: 'center',
    marginBottom: 16,
  },
  scrollView: {
    width: '100%',
    borderColor: '#E0E0E1',
    backgroundColor: '#fff',
    borderRadius: 3,
  },
};

type ScrollViewCardProps = {
  children: any,
};
const ScrollViewCard = ({ children }: ScrollViewCardProps) => (
  <View style={ownStyles.container}>
    <ScrollView
      contentContainerStyle={[ownStyles.scrollView, globalStyles.boxShadow]}
    >
      {children}
    </ScrollView>
  </View>
);

export default ScrollViewCard;
