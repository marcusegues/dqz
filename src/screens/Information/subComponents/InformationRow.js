// @flow
import React from 'react';
// $FlowFixMe
import { TouchableWithoutFeedback, View } from 'react-native';
import Row from '../../../components/Rows/Row';
import { CardRowText } from '../../../components/QuestionAnswer/cards/subcomponents/CardRowText';
import { CardRowSubText } from '../../../components/QuestionAnswer/cards/subcomponents/CardRowSubText';
import GreySquare from './GreySquare';
import RightChevronIcon from '../../../components/General Components/RightChevronIcon';

const ownStyles = {
  container: {
    flexDirection: 'row',
    paddingVertical: 8,
  },
  textContainer: {
    flexDirection: 'column',
    flex: 1,
    justifyContent: 'flex-start',
    paddingLeft: 16,
    paddingRight: 1,
  },
};

type InformationRowProps = {
  mainText: string,
  subText: string,
  number: string,
  rowOnPress: () => void,
};

const InformationRow = ({
  mainText,
  subText,
  number,
  rowOnPress,
}: InformationRowProps) => (
  <Row>
    <TouchableWithoutFeedback onPress={rowOnPress}>
      <View style={ownStyles.container}>
        <GreySquare number={number} />
        <View style={ownStyles.textContainer}>
          <CardRowText text={mainText} style={{ paddingBottom: 3 }} />
          <CardRowSubText text={subText} />
        </View>
        <RightChevronIcon />
      </View>
    </TouchableWithoutFeedback>
  </Row>
);

export default InformationRow;
