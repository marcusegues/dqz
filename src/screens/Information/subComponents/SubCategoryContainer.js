// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { CardHeaderText } from '../../../components/QuestionAnswer/Cards/subcomponents/CardHeaderText';
import { ScrollViewCard } from '../../../components/General Components/ScrollViewCard';
import { MainContentContainer } from '../../../components/MainContentContainer/MainContentContainer';
import { moderateScale } from '../../../styles/Scaling';

type CustomsProceduresInnerProps = {
  children: any,
  title: string,
};

export const SubCategoryContainer = ({
  title,
  children,
}: CustomsProceduresInnerProps) => (
  <MainContentContainer>
    <ScrollViewCard>
      <View style={{ margin: moderateScale(16) }}>
        <CardHeaderText text={title} />
        {children}
      </View>
    </ScrollViewCard>
  </MainContentContainer>
);
