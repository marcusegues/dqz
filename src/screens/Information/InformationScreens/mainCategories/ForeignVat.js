// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
// $FlowFixMe
import type { TFunction } from '../../../../types/generalTypes';
import { verticalScale } from '../../../../styles/Scaling';
import { AppInfoSubText } from '../../../AppInfo/subComponents/AppInfoSubText';
import { SubCategoryContainer } from '../../subComponents/SubCategoryContainer';

type ForeignVatInnerProps = {
  t: TFunction,
};

const ForeignVatInner = ({ t }: ForeignVatInnerProps) => (
  <SubCategoryContainer title={t('foreignVatMainText')}>
    <AppInfoSubText
      text={t('foreignVatText1')}
      style={{ marginTop: verticalScale(10) }}
    />
    <AppInfoSubText
      text={t('foreignVatText2')}
      style={{ marginTop: verticalScale(10) }}
    />
    <AppInfoSubText
      text={t('foreignVatText3')}
      style={{ marginTop: verticalScale(10) }}
    />
    <AppInfoSubText
      text={t('foreignVatText4')}
      style={{ marginTop: verticalScale(10) }}
    />
    <AppInfoSubText
      text={t('foreignVatText5')}
      style={{ marginTop: verticalScale(10) }}
    />
  </SubCategoryContainer>
);

export const ForeignVat = (translate(['foreignVat'])(
  ForeignVatInner
): ComponentType<ForeignVatInnerProps>);
