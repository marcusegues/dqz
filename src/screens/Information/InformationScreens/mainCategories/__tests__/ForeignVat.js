import React from 'react';
// $FlowFixMe
import { Linking, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import { ForeignVat } from '../ForeignVat';
import { AppInfoLink } from '../../../../AppInfo/subComponents/AppInfoLink';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

describe('ForeignVat', () => {
  test('ForeignVat renders as expected', () => {
    const wrapper = shallow(<ForeignVat />);
    expect(wrapper).toMatchSnapshot();
  });
  test('ForeignVat renders as expected with language', () => {
    const wrapper = shallow(
      <ForeignVat
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de', t: 'test' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('ForeignVat renders as expected with language with dive()', () => {
    const wrapper = shallow(
      <ForeignVat
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de', t: 'test' }}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  // test('ForeignVat renders as expected and you can click the button', () => {
  //   const wrapper = shallow(
  //     <ForeignVat
  //       navigation={{ dispatch: clickFn }}
  //       i18n={{ language: 'de', t: 'test' }}
  //     />
  //   );
  //   const render = wrapper.dive();
  //   render
  //     .find(<AppInfoLink title="test" onPress={clickFn} />)
  //     .forEach(child => {
  //       child.simulate('press');
  //     });
  //   expect(clickFn.mock.calls.length).toBe(1);
  //   expect(render).toMatchSnapshot();
  // });
});
