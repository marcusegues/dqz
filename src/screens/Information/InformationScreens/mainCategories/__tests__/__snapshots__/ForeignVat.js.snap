// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForeignVat ForeignVat renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ForeignVat ForeignVat renders as expected and you can click the button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    i18n={
      Object {
        "language": "de",
        "t": "test",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "i18n": Object {
        "language": "de",
        "t": "test",
      },
      "navigation": Object {
        "dispatch": [Function],
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "i18n": Object {
          "language": "de",
          "t": "test",
        },
        "navigation": Object {
          "dispatch": [Function],
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ForeignVat ForeignVat renders as expected with language 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    i18n={
      Object {
        "language": "de",
        "t": "test",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "i18n": Object {
        "language": "de",
        "t": "test",
      },
      "navigation": Object {
        "dispatch": [Function],
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "i18n": Object {
          "language": "de",
          "t": "test",
        },
        "navigation": Object {
          "dispatch": [Function],
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
